gp-res-filter/src/main/java/com/ibm/g11n/pipeline/resfilter/AndroidStringsResource.java

```diff
@@ -150,67 +222,168 @@ public void write(OutputStream os, String language,
     }
 
     @Override
-    public void merge(InputStream base, OutputStream outStream, String language,
-            Map<String, String> data) throws IOException {
-        Scanner in = new Scanner(base, "UTF-8");
-        String line = "";
-        String pattern = "^.*<string.*name=\".*\">.*\n";
+    public void merge(InputStream base, OutputStream outStream, String language, Collection<ResourceString> data)
+            throws IOException {
+        // put res data into a map for easier searching
+        Map<String, String> resMap = new HashMap<String, String>(data.size() * 4 / 3 + 1);
+        for (ResourceString res : data) {
+            resMap.put(res.getKey(), res.getValue());
+        }
 
-        while (in.hasNextLine()) {
-            line = in.nextLine() + "\n";
+        BufferedReader reader = new BufferedReader(new InputStreamReader(base, CHAR_SET));
```

Comment:
i know this isn't a delta, but why is the merge done at the regex level and not the DOM level? It seems like it could be more fragile this way, if someone is doing something strange with the XML such as `<string-array name = "whatever" >`